server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadataBatch.java

```diff
@@ -0,0 +1,535 @@
+        }
+
+        public NodeStoreFilesMetadata(StreamInput in) throws IOException {
+            storeFilesMetadata = new StoreFilesMetadata(in);
+            if (in.readBoolean()) {
+                this.storeFileFetchException = in.readException();
+            } else {
+                this.storeFileFetchException = null;
+            }
+        }
+
+        public NodeStoreFilesMetadata(StoreFilesMetadata storeFilesMetadata, Exception storeFileFetchException) {
+            this.storeFilesMetadata = storeFilesMetadata;
+            this.storeFileFetchException = storeFileFetchException;
+        }
+
+        public StoreFilesMetadata storeFilesMetadata() {
+            return storeFilesMetadata;
+        }
+
+        public static NodeStoreFilesMetadata readListShardStoreNodeOperationResponse(StreamInput in) throws IOException {
+            return new NodeStoreFilesMetadata(in);
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            storeFilesMetadata.writeTo(out);
+            if (storeFileFetchException != null) {
+                out.writeBoolean(true);
+                out.writeException(storeFileFetchException);
+            } else {
+                out.writeBoolean(false);
+            }
+        }
+
+        public Exception getStoreFileFetchException() {
+            return storeFileFetchException;
+        }
+
+        @Override
+        public String toString() {
+            return "[[" + storeFilesMetadata + "]]";
+        }
+    }
+
+    /**
+     * NodeRequest class is for deserializing the  request received by this node from other node for this transport action.
+     * This is used in {@link TransportNodesAction}
+     * @opensearch.internal
+     */
+    public static class NodeRequest extends TransportRequest {
+
+        private final Map<ShardId, String> shardIdsWithCustomDataPath;
+
+        public NodeRequest(StreamInput in) throws IOException {
+            super(in);
+            shardIdsWithCustomDataPath = in.readMap(ShardId::new, StreamInput::readString);
+        }
+
+        public NodeRequest(Request request) {
+            this.shardIdsWithCustomDataPath = Objects.requireNonNull(request.getShardIdsWithCustomDataPath());
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeMap(shardIdsWithCustomDataPath, (o, k) -> k.writeTo(o), StreamOutput::writeString);
+        }
+
+        public Map<ShardId, String> getShardIdsWithCustomDataPath() {
+            return shardIdsWithCustomDataPath;
+        }
+    }
+
+    /**
+     * NodeStoreFilesMetadataBatch Response received by the node from other node for this transport action.
+     * Refer {@link TransportNodesAction}
+     */
+    public static class NodeStoreFilesMetadataBatch extends BaseNodeResponse {
+        private final Map<ShardId, NodeStoreFilesMetadata> nodeStoreFilesMetadataBatch;
```

Comment:
just **storeFilesMetadataBatch** ?

Reply:
I think what @sudarshan-baliga has kept is analogous to Primaries response, where we have prefix of `Node` and in replicas transport(without batch one) we have this prefix.

Reply:
I have kept the naming similar to the non batch version and same is the case with primary transpsort action .